### PR TITLE
Removes major and minor crime distinction, simplifies adding a crime in sechuds

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

Removes minor and major crimes to simplify them, they are just "crimes" now.
Adding a crime is now ~~2~~ 1 pop-up~~s~~ instead of 3.
Instead of adding details in sechuds, you can add it in sec consoles or by viewing crime in hud.

## Why It's Good For The Game

My goal was to simplify adding crimes to encourage security to actually do that.
Since the space law is only a roleplay suggestion, we don't need that distinction.

## Changelog
:cl:
del: Deleted minor and major crimes (they are just crimes now).
tweak: Adding a crime (via sec huds) is now only 1 pop-up, seriously try it.
add: You can now add crime details in security consoles.
/:cl:

### CODE REVIEW

**Comment by improvedname** [comment]
Boo keep major/minor for new players so they can tell whats worser or whats light

**Comment by wesoda25** [comment]
Implying new players would learn how to check security records before the basic rules of the game

**Comment by Archanial** [comment]
They can just use common sense

**Comment by ArcaneMusic** [comment]
Alternatively: Add a prompt when adding a crime to a player to assign a severity level, which beepsky could reflect when arresting that person. (EG: Level 1 Criminal Alert, Level 5, Level 10, etc)

**Comment by Ryll-Ryll** [comment]
All crime is a major crime when you're packing a .357

**Comment by ExcessiveUseOfCobblestone** [comment]
the cursed take is replace with a numerical input for the crime code and have it autopopulate the fields, the officer can edit later 😏

**Comment by Archanial** [comment]
What do you think about stripping details too? It would make adding crime even easier and more convenient. 
When I add crime I just type all the stuffs in its name and nothing in details anyway.

**Comment by cacogen** [comment]
in for a nickel in for a dime

**Comment by cacogen** [comment]
>What do you think about stripping details too?

yes do the multiple pop-up windows are bullshit and should get fucked

**Comment by AdamElTablawy** [comment]
ah yes, delete roleplay from the game, good idea

**Comment by cacogen** [comment]
few people bother spending the time to fill out the records correctly, and it asking (or is it requiring? i know it was at one point) you to fill in two fields often feels redundant and is quite annoying when you're trying to quickly set someone to arrest and then get after them

**Comment by Archanial** [comment]
> 
> 
> few people bother spending the time to fill out the records correctly, and it asking (or is it requiring? i know it was at one point) you to fill in two fields often feels redundant and is quite annoying when you're trying to quickly set someone to arrest and then get after them

It's asking, you don't have to input any details I think.

**Comment by Archanial** [comment]
I got even better idea, just not ask for details while adding crime but make someone be able to input them manually in the security console. Sounds good?

**Comment by cacogen** [comment]
yes

**Comment by necromanceranne** [comment]
I actually categorize my laws between this. I get it's a meme, but if I put shit in 'major' it's because they're almost certainly going into perma.

**Comment by cacogen** [comment]
I do too, as well as try to fill out things in detail (you still get geniuses asking over the radio why someone is set to arrest though). But let me just put in the crime name and fill in the details later

**Comment by Bawhoppen** [comment]
Cool

File: code/game/machinery/computer/security.dm [hunk 1]
```diff
@@ -226,32 +226,16 @@
...
-							dat += "<tr><td>[c.crimeName]</td>"
-							dat += "<td>[c.crimeDetails]</td>"
+							if(!c.crimeDetails)
+								dat += "<td><A href='?src=[REF(src)];choice=Edit Field;field=add_details;cdataid=[c.dataId]'>\[+\]</A></td>"
+							else
+								dat += "<td>[c.crimeDetails]</td>"
 							dat += "<td>[c.author]</td>"
 							dat += "<td>[c.time]</td>"
-							dat += "<td><A href='?src=[REF(src)];choice=Edit Field;field=ma_crim_delete;cdataid=[c.dataId]'>\[X\]</A></td>"
+							//dat += "<td><A href='?src=[REF(src)];choice=Edit Field;field=add_details;cdataid=[c.dataId]'>\[X\]</A></td>"
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/machinery/computer/security.dm]: commented code

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -450,34 +443,18 @@
...
-						if(!H.canUseHUD())
-							return
-						if(!HAS_TRAIT(H, TRAIT_SECURITY_HUD))
-							return
-						var/crime = GLOB.data_core.createCrimeEntry(t1, t2, allowed_access, station_time_timestamp())
-						GLOB.data_core.addMajorCrime(R.fields["id"], crime)
-						investigate_log("New Major Crime: <strong>[t1]</strong>: [t2] | Added to [R.fields["name"]] by [key_name(usr)]", INVESTIGATE_RECORDS)
-						to_chat(usr, "<span class='notice'>Successfully added a major crime.</span>")
+				var/t1 = stripped_input("Please input crime name:", "Security HUD", "", null)
+				var/t2 = null
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/carbon/human/human.dm]: what does this do?

**Review by ExcessiveUseOfCobblestone — COMMENTED** [review]
if you are familiar enough with it please autodoc the procs touched too.

File: code/modules/mob/living/carbon/human/human.dm (hunk 2 above)
Comment (Archanial) [reply code/modules/mob/living/carbon/human/human.dm]: t2 was used for details, just forgot to remove it afterwards

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/carbon/human/human.dm]: need to fix

Comment (Archanial) [reply code/modules/mob/living/carbon/human/human.dm]: I'll push the changes tomorrow

File: code/datums/datacore.dm [hunk 3]
```diff
@@ -75,38 +75,59 @@
...
-/datum/datacore/proc/addMinorCrime(id = "", datum/data/crime/crime)
+/**
+  * Adds crime to security record.
+  *
+  * Is used to add single crime to someone's security record.
+  * Arguments:
+  * * id - record id.
+  * * datum/data/crime/crime - premade array containing every variable, usually created by createCrimeEntry.
+  */
+
```
Comment (optimumtact) [inline code/datums/datacore.dm]: no space

Comment (optimumtact) [reply code/datums/datacore.dm]: same goes for rest

**Pull request merged**